addons/project/models/project.py

```diff
@@ -1395,6 +1400,19 @@ def _compute_stage_id(self):
             else:
                 task.stage_id = False
 
+    @api.depends('user_ids')
+    def _compute_portal_user_ids(self):
+        tasks_search_read = self.search_read([('id', 'in', self.ids), ('user_ids', '!=', False)], ['id', 'user_ids'])
+        assignees_by_task = {res['id']: res['user_ids'] for res in tasks_search_read}
+        for task in self:
+            task.portal_user_ids = [Command.set(assignees_by_task.get(task.id, []))]
```

Comment:
Why not just e.g.
```suggestion
        for task in self:
            task.portal_user_ids = [Command.set(task.user_ids.ids)]
```
? I don't think all the complexity is needed here, we're just reading *one* field.

Reply:
I cannot do that because the `user_ids` is empty when I am a portal user since he cannot access to `res.users`. Perhaps I am wrong but I think we use the cache and so we don't call the db in sudo to have the real assignees in the task.

Reply:
What about this?
```suggestion
        for task in self:
            task.portal_user_ids = task.sudo().user_ids._filter_access_rules_python('read')
```

Reply:
It seems it doesn't work for the same reason. 😢

Reply:
at that point this is just a `fields.Many2Many(related='user_ids')`, which defaults to `related_sudo` aka `compute_sudo=True`, no?

except that the cache is probably populated in a non-sudo env first, so when accessed with `related_sudo` it doesn't retrieve any extra content. Perhaps that's what @xavierbol is referring to?

Reply:
Yeah, indeed, it is exactly this, I wanted to say.